fhir-operation-bulkdata/src/main/java/com/ibm/fhir/operation/bulkdata/util/BulkDataUtil.java

```diff
@@ -284,4 +287,37 @@ public static String checkAndValidateJob(Parameters parameters) throws FHIROpera
 
         throw new FHIROperationException("no job identifier is passed");
     }
+
+
+    public static String encryptBatchJobId(String strToEncrypt, SecretKeySpec key) {
+        // Encrypt and UrlEncode the batch job id.
+        if (key == null) {
+            return strToEncrypt;
+        } else {
+            try
+            {
+                Cipher cp = Cipher.getInstance("AES/ECB/PKCS5Padding");
+                cp.init(Cipher.ENCRYPT_MODE, key);
+                return java.net.URLEncoder.encode(Base64.getEncoder().encodeToString(cp.doFinal(strToEncrypt.getBytes("UTF-8"))), StandardCharsets.UTF_8.name());
+            } catch (Exception e) {
+                return strToEncrypt;
+            }
+        }
+    }
+
+    public static String decryptBatchJobId(String strToDecrypt, SecretKeySpec key) {
+        // Decrypt to get the batch job id
+        if (key == null) {
+            return strToDecrypt;
+        } else {
+            try
+            {
+                Cipher cp = Cipher.getInstance("AES/ECB/PKCS5PADDING");
```

Comment:
is this the lightest weight cipher?  it doesn't need to be strong.

Reply:
Where is the URL decode? shouldn't this be symmetric, or are you relying on the Liberty runtime to previously decode

Reply:
maybe add a comment

Reply:
yes, this is lightest weight cipher, and I didn't add salt to it to make the config simpler, otherwise, we will need config for both the key and salt.
yes, liberty run time or the rest framework did the url decoding automatically, I added decoding in this function before, and found it didn't work, and then through debugging, I found it's already decoded, double decode the encrypted job id caused error, so I removed the url decode from this function.  
Yes, will add comment for this.

Reply:
great confirmation, I think adding the comment is the only part left?

Reply:
added commenats